### ISSUE TITLE
Add possibility to change lexer settings directly in Notepad++

### ISSUE THREAD

**Issue by ArkadiuszMichalski** [notepad-plus-plus/notepad-plus-plus#12213]
Some lexers have their own options (mostly for fold/unfold, but not only). It would be nice if we could change these options directly in Notepad++ (without additional plugins like NppExec or PythonScript). Not everyone have or can install these plugins (also know how to use them).

This can probably be done in many ways, these are just two suggestions:
- via GUI in the Preferences window. Show a list of lexers, each has its own text field where we enter the appropriate settings (each option on a separate line), e.g.
```
lexer.css.less.language=1
lexer.css.scss.language=1
```
This change can be saved by using the `xml` file below.
- via `langs.model.xml / langs.xml` file. For each `<Language>` we can add additional tag for configuration, like `<settings> or <config>` or other name. It contains another tag representing a specific option, like `<option>`. So for example:
```
<Language name="css" ext="css" commentStart="/*" commentEnd="*/">
    ....
    <settings>
        <option>lexer.css.less.language=1</option>
        <option>lexer.css.scss.language=1</option>
    </settings>
    ....
```

If user has made such changes (its lexer settings are different from the default ones) it's worth adding some info about it in Debug info. Enough is listing such changed lexers, for example:
```
Modified lexers: L_JS, L_CSS
```

I will just add that there are some open issues where we can solve them by using a specific option of a given lexer. Unfortunately, in Notepad++ only one such option is provided directly via the GUI for  escape `\` SQL. Selectively adding options is pointless, it should be possible to change any of them (as above).

Any suggestions are welcome.

**Comment by niekma** [comment]
May I ask why you would like to have separate areas for each lexer?
I think all options have different names, so it would be sufficient to have one place (e.g. Advanded ScintillaEditView Settings).
This could be a simple TextArea in the preferences window of the GUI, for example.

**Comment by ArkadiuszMichalski** [comment]
@niekma 
>I think all options have different names, so it would be sufficient to have one place

No, names can be very different (there is no specific pattern), in addition they are also the same for different lexers. You can check this two files by searching for `execute(SCI_SETPROPERTY`:
https://github.com/notepad-plus-plus/notepad-plus-plus/blob/83ee8f09bf84980809d7b58e008314b4672566e7/PowerEditor/src/ScintillaComponent/ScintillaEditView.h
https://github.com/notepad-plus-plus/notepad-plus-plus/blob/4b3ee9c4eae5b3ca0412aec0e746083a9a7d6317/PowerEditor/src/ScintillaComponent/ScintillaEditView.cpp

There will be no confusion if each lexer has separate settings. In fact, we already have everything we need, because structure of `langs.model.xml/langs.xml` looks like this. UI appearance is a secondary case (for discussion). Either way, we have to remember these  settings somehow, and mentioned XML files are perfect for such purpose. It might even be without UI at first, it's still better than having no control over it at all (of course without the use of plugins).

But at the very beginning @donho must accept this type of change anyway to avoid potential rejection of any work.

**Comment by niekma** [comment]
@ArkadiuszMichalski: Thanks for the explanation and the source code hints.

**Comment by Fulmir** [comment]
Just want to second this, after spending about 2 hours poking around trying to figure out how to do exactly this.

There's already even code in each Lexer with descriptions for the settings, and they're set in the tests so it's possible to do so.

This literally just needs a settings page, a list of Lexers in a scrolling list, and then a dynamic UI element to list each setting with an edit field.

Heck, might even look into doing this myself sometime, since there's apparently already a setting for this in SQL I can use as reference...
